pkg/agent/agent_metrics_test.go

```diff
@@ -46,33 +48,44 @@ func agentMetricsSetupTest() (*MockClient, *Agent) {
 		comm: &RPCComm{
 			client: clientMock,
 		},
+		metricStack: map[string][]*io_prometheus_client.MetricFamily{},
 	}
 
 	return clientMock, stubbedAgent
 }
 
 func TestMetricsRelayHandler(t *testing.T) {
 
-	t.Run("will relay the metrics", func(t *testing.T) {
+	t.Run("will relay metrics from the stack", func(t *testing.T) {
```

Comment:
Two test cases that I think still need coverage are:

1. backoff time was met but no metrics available in `metricStack`
2. unable to get peer IP address from context, thus no metrics sent